buildSrc/src/main/kotlin/options.kt

```diff
@@ -0,0 +1,33 @@
+import org.gradle.api.Project
+import org.gradle.api.plugins.JavaPluginExtension
+import org.gradle.api.tasks.compile.JavaCompile
+import org.gradle.jvm.toolchain.JavaLanguageVersion
+import org.jetbrains.kotlin.gradle.dsl.KotlinJvmOptions
+import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
+
+
+fun Project.configureCompilerOptions(jvmVersion: Int = 8) {
+    tasks.withType(KotlinCompile::class.java).configureEach {
+        it.compilerOptions {
+            //allWarningsAsErrors.set(true)
```

Comment:
I commented this for now as it's a bit painful during development (unused variables, etc...) I'll enable it from time to time keep warnings in check